File: catalog/backends/ruby-rails/patterns/cpu_intensive_service.rb
# Pattern: Profiling — CPU-intensive nested loop
# Adapt: replace with domain-relevant aggregation logic

def self.aggregate
  counts = {}
  100.times do
    counts.clear
    10_000.times do |j|
      key = "bucket-#{j % 50}"
      counts[key] = (counts[key] || 0) + 1
    end
  end
  { buckets: counts.size, iterations: 100 * 10_000 }
end

File: catalog/backends/java-spring/patterns/CpuIntensiveService.java
package com.example.service;

import org.springframework.web.bind.annotation.*;
import java.util.*;

// Pattern: Profiling — CPU-intensive nested loop
// Adapt: replace with domain-relevant aggregation logic
@RestController
public class CpuIntensiveService {

    @GetMapping("/api/aggregate")
    public Map<String, Object> aggregate() {
        Map<String, Long> counts = new HashMap<>();
        for (int i = 0; i < 100; i++) {
            counts.clear();
            for (int j = 0; j < 10000; j++) {
                String key = "bucket-" + (j % 50);
                counts.merge(key, 1L, Long::sum);
            }
        }
        return Map.of("buckets", counts.size(), "iterations", 100 * 10000);
    }
}
